### PR TITLE
RK-2652 Analytics

### PR DESCRIPTION
**Pull request by amits1995**

### CODE REVIEW

File: src/index-worker.ts [hunk 1]
```diff
@@ -1,39 +1,69 @@
...
 
 const getRepos = () => repStore.getRepositories().map((r) => r.toModel());
 
+const isPortInUse = (port: number): Promise<boolean> => new Promise<boolean>((resolve, reject) => {
+    const testServer = net.createServer()
+    .on("error", (err: any) => {
+        err.code === "EADDRINUSE" ? resolve(true) : reject(err);
+    })
+    .on("listening", () => {
+        testServer.once("close", () => resolve(false));
```
Comment (MickaelAlliel) [inline src/index-worker.ts]: What happens if you cant close the server? You're doing the same mistake they did

File: src/index.ts [hunk 2]
```diff
@@ -147,19 +182,19 @@ function main() {
...
-    if (!token) {
+    token = store.getOrCreate("token", uuidv4(), () => {
         firstTimeLaunch = true;
-        token = uuidv4();
-        store.set("token", token);
+    });
+    userId = store.getOrCreate("user-id", uuidv4());
+    exceptionManagerEnabled = store.get("sentry-enabled", true);
+    const signedEula = store.get("has-signed-eula", false);
+    if (signedEula && exceptionManagerEnabled && !process.env.development) {
```
Comment (MickaelAlliel) [inline src/index.ts]: we may want to rename `exceptionManagerEnabled` to something like `allowCollectingData`

File: src/middlewares.ts [hunk 3]
```diff
@@ -69,10 +69,13 @@ export const authenticateController: AuthenticateController = (token) => {
       return;
     }
     const domain: string = envDict.get(env);
-    const targetUrl = `${domain}/authorize/explorook#token=${token}`;
+    const targetUrl = `${domain}/authorize/explorook?user-id=${userId}#token=${token}`;
+    ipcRenderer.send("track", "authorize-open-chrome");
```
Comment (MickaelAlliel) [inline src/middlewares.ts]: not sending data? userid?

File: src/explorook-store.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import Store = require("electron-store");
+
+export class ExplorookStore extends Store {
+    constructor(name: string = "explorook") {
+        super({
+            name
+        });
+    }
+
+    public getOrCreate(key: string, value: any, onCreated: () => void = null): any {
```
Comment (MickaelAlliel) [inline src/explorook-store.ts]: we dont need `update`?

**Comment by MickaelAlliel** [comment]
@amits1995 Still have a snyk error as well

File: src/index-worker.ts (hunk 1 above)
Comment (amits1995) [reply src/index-worker.ts]: What mistake? who did it?  
If the `'listening'` event is triggered, the server is running and can be closed.  
If there is an error with the close operation - the error event will trigger

File: src/index.ts (hunk 2 above)
Comment (amits1995) [reply src/index.ts]: OK

File: src/explorook-store.ts (hunk 4 above)
Comment (amits1995) [reply src/explorook-store.ts]: no, I just extended the store with this helper method (I found myself needing this alot)

File: src/middlewares.ts (hunk 3 above)
Comment (amits1995) [reply src/middlewares.ts]: solved f2f

**Pull request merged**